### PR TITLE
NF: uses kotlin.test instead of Matcher and Junit

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This ensure that the value is non null by contract, so we can remove ulterior !!

Replace and improve #12352

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]
I really like the changes. The contracts help very much here.

**Comment by BrayanDSO** [comment]
I believe that https://github.com/ankidroid/Anki-Android/blob/main/lint-rules/src/main/java/com/ichi2/anki/lint/rules/JUnitNullAssertionDetector.kt isn't working as expected. If it were the case, these changes should have been flagged by lint ages ago

**Review by david-allison — CHANGES_REQUESTED** [review]
Strong disagree (as listed in the PR this supersedes): we want one consistent method to do assertions.

Making it inconsistent in one of our most important test classes isn't a good idea

**Comment by BrayanDSO** [comment]
@david-allison, in your goal of consistency, is the approach of https://github.com/ankidroid/Anki-Android/pull/12352#issuecomment-1242825257 fine?

I believe that both kotlin.test and kotlin.test.junit both have contracts

**Comment by Arthur-Milchior** [comment]
Before https://github.com/ankidroid/Anki-Android/pull/12352  was merged, the IDE noted that this `assertNotNull` was used 80 times, so I thought I was being consistent. I now understand the IDE was wrong, and since we don't use this anywhere, I now agree with your note that my implementation was not right

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt [hunk 1]
```diff
@@ -40,6 +40,6 @@ class CollectionTest : InstrumentedTest() {
     @Test
     @KotlinCleanup("use the recommended assertion to remove the @SuppressLint")
     fun testOpenCollection() {
-        assertNotNull("Collection could not be opened", col)
+        assertNotNull(col, "Collection could not be opened")
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/CollectionTest.kt:43]: We do not want to introduce an ambiguous manner of specifying the message for a test. 

Historically the message has been first, and we want one convention or the other.

With this, if `col` was a string, both `assertNotNull(col, "message")` and `assertNotNull("message", col)` would compile

I don't want to have to think about which one is correct.

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by david-allison** [comment]
> @david-allison, in your goal of consistency, is the approach of https://github.com/ankidroid/Anki-Android/pull/12352#issuecomment-1242825257 fine?
> 
> I believe that both kotlin.test and kotlin.test.junit both have contracts, and as the junit module delegates to the same asserter of junit, shouldn't be a problem 

I don't think I described consistency, I've elaborated in a review comment

**Comment by david-allison** [comment]
Can we take this back to a GitHub issue or Discord? 

This is a massive undertaking. I'd struggle to find justification to accept the PR, and I don't feel this is a productive use of time.

**Pull request closed without merging**